File: src/UE4_Packages/FGenerationInfo.java
/**
 * 
 */
package UE4_Packages;

/**
 * @author FunGames
 *
 */
public class FGenerationInfo {
	private int exportCount;
	private int nameCount;
	
	
	public FGenerationInfo(FArchive Ar) throws ReadException {
		exportCount = Ar.readInt32();
		nameCount = Ar.readInt32();
	}


	public int getExportCount() {
		return exportCount;
	}

	public int getNameCount() {
		return nameCount;
	}
	
	
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/assets/objects/FGenerationInfo.kt
package me.fungames.jfortniteparse.ue4.assets.objects

import me.fungames.jfortniteparse.ue4.UClass
import me.fungames.jfortniteparse.ue4.reader.FArchive
import me.fungames.jfortniteparse.ue4.writer.FArchiveWriter

@ExperimentalUnsignedTypes
class FGenerationInfo : UClass {
    var exportCount: Int
    var nameCount: Int

    constructor(Ar: FArchive) {
        super.init(Ar)
        exportCount = Ar.readInt32()
        nameCount = Ar.readInt32()
        super.complete(Ar)
    }

    fun serialize(Ar: FArchiveWriter) {
        super.initWrite(Ar)
        Ar.writeInt32(exportCount)
        Ar.writeInt32(nameCount)
        super.completeWrite(Ar)
    }

    constructor(exportCount: Int, nameCount: Int) {
        this.exportCount = exportCount
        this.nameCount = nameCount
    }
}